### PR TITLE
fix: support consecutive capital letters

### PR DESCRIPTION
**Pull request by bcoe**
The simple algorithm I wrote for decamelize did not support acronyms well (the bug is [seen here](https://github.com/googleapis/nodejs-bigquery/blob/45b6f7634e6517ca217c0d73b59b818256549750/README.md)).

This PR extends the method adding this functionality and adds tests for this edge-case; CC: @steffnay

fixes: #219

### CODE REVIEW

File: synthtool/__main__.py [hunk 1]
```diff
@@ -70,8 +70,8 @@ def main(synthfile: str, metadata: str, extra_args: Sequence[str]):
     if os.path.lexists(synth_file):
         synthtool.log.debug(f"Executing {synth_file}.")
         # https://docs.python.org/3/library/importlib.html#importing-a-source-file-directly
-        spec = importlib.util.spec_from_file_location("synth", synth_file)
-        synth_module = importlib.util.module_from_spec(spec)
+        spec = util.spec_from_file_location("synth", synth_file)
```
Comment (bcoe) [inline synthtool/__main__.py]: I was receiving linting errors locally without switching to this approach (upstream regression perhaps?).

File: synthtool/gcp/common.py [hunk 2]
```diff
@@ -153,10 +153,22 @@ def _load_partials(self, metadata: Dict):
 
 def decamelize(str: str):
     """ parser to convert fooBar.js to Foo Bar. """
-    str2 = str[0].upper()
-    for chr in str[1:]:
-        if re.match(r"[A-Z]", chr):
-            str2 += " " + chr.upper()
+    prev = " "
+    str2 = ""
```
Comment (tswast) [inline synthtool/gcp/common.py]: I see this is existing code, but `output` or `decamelized` would be much more readable than `str2`.

File: synthtool/gcp/common.py [hunk 3]
```diff
@@ -153,10 +153,22 @@ def _load_partials(self, metadata: Dict):
...
+    for i in range(0, len(str)):
```
[the remark this answers is not in the document]
Comment (tswast) [reply synthtool/gcp/common.py]: Also, `chr` is a built-in, so it's best to avoid that as a variable name. Maybe `current` to correspond to `prev`?

File: synthtool/gcp/common.py [hunk 4]
```diff
@@ -153,10 +153,22 @@ def _load_partials(self, metadata: Dict):
...
+        chr = str[i]
+        if prev == " ":
+            str2 += chr.upper()
+        elif re.match(r"[A-Z]", chr):
+            # handle "JSONFrom" -> "JSON From" case.
+            next = ""
```
Comment (tswast) [inline synthtool/gcp/common.py]: `next` is also a built-in, so it's best not to overwrite it. Usually we use a trailing underscore to avoid these kinds of collisions. `next_`.

File: synthtool/gcp/common.py [hunk 5]
```diff
@@ -153,10 +153,22 @@ def _load_partials(self, metadata: Dict):
 
 def decamelize(str: str):
```
Comment (tswast) [inline synthtool/gcp/common.py]: `str` is a built-in, so we probably shouldn't use it as a variable name. How about `value` or `str_`?

File: synthtool/gcp/common.py [hunk 6]
```diff
@@ -153,10 +153,22 @@ def _load_partials(self, metadata: Dict):
...
+            next = ""
+            if i < len(str) - 1:
+                next = str[i + 1]
+            # "CoolACL" -> "Cool ACL" case.
+            if not re.match(r"[A-Z]", prev) or re.match(r"[a-z]", next):
+                str2 += " "
+            str2 += chr
         else:
             str2 += chr
+        prev = str2[len(str2) - 1]
```
Comment (tswast) [inline synthtool/gcp/common.py]: I guess this is to handle when `prev` should be `" "`? I'd prefer if the above set a temporary variable for the string to append, then this becomes

```
output += to_append
prev = to_append
```

File: synthtool/__main__.py (hunk 1 above)
Comment (tswast) [reply synthtool/__main__.py]: Honestly that should be a linting error. It's not always guaranteed that submodules are actually loaded when you import a top-level module.

That said, I'd prefer that you instead change the above import to `import importlib.util` instead. `util` is too common of a name to be meaningful.
